srw/srw.cpp

```diff
@@ -74,17 +89,60 @@ class dnet_upstream_t: public cocaine::api::stream_t
 		}
 
 		virtual void write(const char *chunk, size_t size) {
-			reply(false, chunk, size);
+			int err = 0;
+			/* Chunk should be decoded */
+			msgpack::object obj;
+			size_t offset = 0;
+
+			msgpack::unpack_return rv = msgpack::unpack(chunk, size, &offset, &m_zone, &obj);
+			if (rv != msgpack::UNPACK_SUCCESS && rv != msgpack::UNPACK_EXTRA_BYTES) {
+				err = EINVAL;
+				SRW_LOG(*m_node->log, DNET_LOG_ERROR, "app/" + m_sph.event, "unable to unpack response");
+			} else if (obj.type != msgpack::type::RAW) {
+				err = EINVAL;
+				SRW_LOG(*m_node->log, DNET_LOG_ERROR, "app/" + m_sph.event, "insorrect response type, should be string, got: %d", obj.type);
+			}
+
+			if (err) {
+				error(err, "protocol error");
```

Comment:
Should it be more descriptive message?

Also, why does it use SRW_LOG directly and error() wrapper simultaneously?